agent/lib_aws_sdk_php.c

```diff
@@ -0,0 +1,170 @@
+#define PHP_AWS_CLASS_PREFIX "Supportability/library/aws/aws-sdk-php/"
+
+/*
+ * In a normal course of events, the following line will always work
+ * zend_eval_string("Aws\\Sdk::VERSION;", &retval, "Get AWS Version")
+ * By the time we have detected the existence of the aws-sdk-php and with
+ * default composer profject settings, it callable even from
+ * nr_aws_sdk_php_enable which will automatically load the class if it isn't
+ * loaded yet and then evaluate the string. However, in the rare case that files
+ * are not loaded via autoloader and/or have non-default composer classload
+ * settings, if the class is not found, PHP 8.2+ will generate a fatal
+ * unrecoverable uncatchable error error whenever it cannot find a class. While
+ * calling this from nr_aws_sdk_php_enable would have been great and would allow
+ * the sdk version value to be set only once, to avoid the very unlikely but not
+ * impossible fatal error, this will be called from the
+ * "Aws\\ClientResolver::_apply_user_agent" wrapper which GUARANTEES that
+ * aws/sdk exists and is already loaded.
+ *
+ *
+ * Additionally given that aws-sdk-php is currently detected from the
+ * AwsClient.php file, this method will always be called when a client is
+ * created. Having separate functionality to extract from Aws/Sdk::__constructor
+ * is both not required and is redundant and causes additional overhead.
+ */
+extern void lib_aws_sdk_php_handle_version() {
+  char* version = NULL;
+  zval retval;
+  int result = FAILURE;
+
+  result = zend_eval_string("Aws\\Sdk::VERSION;", &retval, "Get AWS Version");
+
+  // Add php package to transaction
+  if (SUCCESS == result) {
+    if (nr_php_is_zval_non_empty_string(&retval)) {
+      version = Z_STRVAL(retval);
+    }
+    zval_dtor(&retval);
+  }
+
+  /* Use the Aws\Sdk::VERSION to determine the version */
+  if (NRINI(vulnerability_management_package_detection_enabled)) {
+    /* Add php package to transaction */
+    nr_txn_add_php_package(NRPRG(txn), PHP_PACKAGE_NAME, version);
+  }
+  nr_fw_support_add_package_supportability_metric(NRPRG(txn), PHP_PACKAGE_NAME,
+                                                  version);
+}
+
+extern void lib_aws_sdk_php_add_supportability_metric(const char* metric_name) {
+  int MAX_LEN = 512;
+  char buf[MAX_LEN];
+  nrobj_t* names = NULL;
+  char* DELIM = "_";
+
+  if (NULL == metric_name) {
+    return;
+  }
+  if (NULL == NRPRG(txn)) {
+    return;
+  }
+
+  /* Strip out the `\\` characters and normalize */
+  names = nr_strsplit(metric_name, "\\", 0 /* discard empty */);
+
+  buf[0] = '\0';
+
+  int cur_len = snprintf(buf, sizeof(buf), "%s%s", PHP_AWS_CLASS_PREFIX,
+                         nro_get_array_string(names, 1, NULL));
+  for (int i = 2, n = nro_getsize(names); i <= n; i++) {
+    const char* name = nro_get_array_string(names, i, NULL);
+    if (NULL != name) {
+      cur_len
+          += snprintf(buf + cur_len, MAX_LEN - cur_len, "%s%s", DELIM, name);
+    }
```

Comment:
Can we break out early here if `cur_len >= MAX_LEN`?

Reply:
good idea, please see: [3b34325](https://github.com/newrelic/newrelic-php-agent/pull/932/commits/3b343255641d44ec4a4b6c5bab37499052b497f0)